### PR TITLE
Fixes Brave ads are blocked forever after ads limits (per hour or per day) are reached

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/15814

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See https://github.com/brave/brave-browser/issues/15814

### CODE REVIEW

File: vendor/bat-native-ads/src/bat/ads/internal/ad_events/ad_events.cc [hunk 1]
```diff
@@ -69,11 +69,9 @@ void RecordAdEvent(const AdEventInfo& ad_event) {
   const std::string confirmation_type_as_string =
       std::string(ad_event.confirmation_type);
 
-  const base::Time time = base::Time::Now();
-  const uint64_t timestamp = static_cast<uint64_t>(time.ToDoubleT());
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/ad_events/ad_events.cc:73]: When rebuilding ad events after a browser restart all timestamps were incorrectly set to the current timestamp, but should have been set to the ad event timestamp

**Review by moritzhaller — APPROVED** [review]
++

**Comment by tmancey** [comment]
CI failed due to unrelated reasons

**Comment by btlechowski** [comment]
Good to uplift


Verified `Brave Ads maximum new tab page ads per hour - 1 per hour - limit expired`

<details>
Verified 1st ad is shown:
```
[15412:14198:0518/022425.537737:VERBOSE6:new_tab_page_ad_event_viewed.cc(32)] Successfully logged new tab page ad viewed event
```

Verified next ad is blocked:
```
[14198:14198:0518/022429.602785:VERBOSE2:permission_rule_util.cc(23)] You have exceeded the allowed new tab page ads per hour
[14198:14198:0518/022429.602823:VERBOSE1:new_tab_page_ad.cc(120)] New tab page ad: Not allowed
```

After 1h, verified ads are shown again:
```
[15412:15412:0518/032552.756779:VERBOSE6:new_tab_page_ad_event_viewed.cc(32)] Successfully logged new tab page ad viewed event
```

</details>


Verified `Brave Ads maximum ad notifications per day - multiple profiles - 1 per day - limit expired`

<details>
Verified 1st ad is shown:
```
[21591:21591:0518/025148.327830:VERBOSE1:ad_notification_serving.cc(105)] Ad notification delivered:
  uuid: 0ebb3619-bb96-44b0-a8ae-9632de538ac6
  creativeInstanceId: 54592ab6-5a8e-42e6-8e11-9e9c04e9eecb
  creativeSetId: c7e7a381-4c74-4dd9-9b6d-4edf9330454b
  campaignId: 0d01425b-c970-4a25-b2d0-ccf748cfaf6e
  advertiserId: ecbcc833-8b1d-4867-98f7-ad2341396ce8
  segment: untargeted
  title: Ad2 - same set
  body: Ad2 - same set
  targetUrl: https://youtube.com
```

Verified next ad is blocked:
```
[21591:21591:0518/025228.535362:VERBOSE2:permission_rule_util.cc(23)] You have exceeded the allowed ads per day
[21591:21591:0518/025228.535503:VERBOSE1:ad_notification_serving.cc(150)] Ad notification not served: Not allowed
[21591:21591:0518/025228.535601:VERBOSE1:ad_notification_serving.cc(91)] Ad notification not delivered

```

After 24h, verified ads are shown again:
```
[22709:22709:0519/025348.196800:VERBOSE1:ad_notification_serving.cc(105)] Ad notification delivered:
  uuid: e36e7a5d-8a06-4220-8223-f1eb14a0bfe4
  creativeInstanceId: 5b4cb3f4-5285-4597-87c5-3082fe37938a
  creativeSetId: c7e7a381-4c74-4dd9-9b6d-4edf9330454b
  campaignId: 0d01425b-c970-4a25-b2d0-ccf748cfaf6e
  advertiserId: ecbcc833-8b1d-4867-98f7-ad2341396ce8
  segment: untargeted
  title: Ad1 - same set
  body: Ad1 - same set
  targetUrl: https://youtube.com

```

</details>


Verified `Brave Ads maximum new tab page ads per day - multiple profiles - 1 per day - limit expired`
<details>

Verified 1st ad is shown:
```
[19030:19030:0518/024315.423275:VERBOSE6:new_tab_page_ad_event_viewed.cc(32)] Successfully logged new tab page ad viewed event
```
Verified next ad is blocked in 2nd profile:
```
[19030:19030:0518/024546.239251:VERBOSE2:permission_rule_util.cc(23)] You have exceeded the allowed new tab page ads per day
[19030:19030:0518/024546.239289:VERBOSE1:new_tab_page_ad.cc(120)] New tab page ad: Not allowed
```


After 24h, verified ads are shown again:
```
[20259:20259:0519/024640.809757:VERBOSE6:new_tab_page_ad_event_viewed.cc(32)] Successfully logged new tab page ad viewed event
```

</details>

**Pull request merged**